src/frontend/src/views/Upload/components/Samples/utils.ts

```diff
@@ -129,27 +129,49 @@ function handleFastaText(text: string, filename: string): ParseOutcome {
       continue;
     }
 
-    // Aggregate sequence line by line
-    for (let j = i + 1; j < lines.length; j++) {
-      const jLine = lines[j];
+    const { newIndex, sequence } = aggregateSequence(i, lines);
 
-      if (!jLine || jLine.includes(">")) {
-        i = j;
-        break;
-      }
+    i = newIndex;
 
+    if (sequence) {
       const id = iLine.replace(">", "");
 
       result = {
         ...result,
-        [id]: (result[id] || "") + jLine,
+        [id]: sequence,
       };
     }
   }
 
   return { errors, result };
 }
 
+function aggregateSequence(
+  i: number,
+  lines: string[]
```

Comment:
why `string[]` and not `string` here?

Reply:
oh because it's an array of strings?

Reply:
Yup! We break fasta file into a collection of strings via line break `const lines = text.split(/\r?\n/)`, so each string in the array represents a line in the fasta file. And normally line break is just `\n` in macOS, but Windows could use `\r\n`, thus taking that into account in the `.split()` method!